docs/Features.md

```diff
@@ -0,0 +1,216 @@
+
+## Macros
+
+Macros are what enables the literate program to be written in logical order for the human reader.
+Using Yarner, this is accomplished by naming the code blocks, and then referencing them later by
+"invoking" the macro. While the syntax for naming a code block is specific to the documentation
+style, macro invocation is the same.
+
+By default, macro invocations start with a long arrow `==>` and end with a period `.`. Both of these
+sequences can be customized to suit your needs better. The only restriction with macro invocations
+is that they must be the only thing on the line. That is, this is valid:
+
+```rs
+fn main() {
+  ==> Calculate the very complex result.
+  ==> Print the results for the user.
+}
+```
+
+But this would not invoke the macro named within the `if()` as the macro sequences do not start and
+end the line:
+
+```rs
+fn main() {
+  if (==> A very complex condition is true.) {
+    ==> Do something cool.
+  }
+}
+```
+
+Another feature of macros to note is that if two code blocks have the same name, they are
+concatenated, in the order they are written. This can be very useful in defining global variables or
+listing imports closer to the parts where they are used.
+
+## Meta variables
+
+If you consider a macro invocation like a function call, then meta variables are like parameters.
+
+By default, to indicate that a macro includes a meta variable, the name of the variable must be part
+of the name of the macro, delimited by `@{` and `}`.
+
+Then that meta variable may be used within the macro by again using its name within the `@{` and `}`
+in the code.
+
+Finally, a macro with meta variables is invoked by replacing the name of the variable with its
+value in the invocation.
+
+An example:
+
+~~~md
+Here is our macro with meta variables:
+
+```rust - Say @{something} to @{someone}
+println!("Hey, @{someone}! I was told to tell you \"@{something}\"");
+```
+
+Now, to say things to many people:
+
+```rust
+==> Say @{Hello} to @{Jim}.
+==> Say @{How are you} to @{Tom}.
+==> Say @{I am good!} to @{Angela}.
+```
+~~~
+
+Meta variables can have default values:
+
+~~~md
+Here is our macro with default meta variables:
+
+```rust - Say @{something:Hello} to @{someone}
+println!("Hey, @{someone}! I was told to tell you \"@{something}\"");
+```
+
+Now, to say the default "Hello" to many people:
+
+```rust
+==> Say @{} to @{Jim}.
+==> Say @{} to @{Tom}.
+==> Say @{} to @{Angela}.
+```
+~~~
+
+These features allows for more flexibility when writing macros, as well as possibly making the intent
+clearer.
+
+## Extracted comments
+
+By default, the comment extraction sequence is set to `//`, purely for familiarity. Any text after
+(and including) this sequence is extracted from the code block, and not rendered to the tangled
+source code. Note that, since the comments are removed completely when compiling, they do not have
+to use the actual line comment indicator from you programming language. In fact, it may be better to
```

Comment:
I find that questionable, as the generated source code is not intended to be human-readable, so why should it contain comments? The document itself should do all the commentary necessary.

Reply:
Maybe you misunderstand that? Which means it should be expressed more clearly...

I read your comment as "//-Comments in code blocks should go into the code", as they actually do not.

Otherwise, I can only interprete it as "strip all comments out" (incl. the language's "real" comments).

Reply:
I am just referring to the last part about it being better to have comments in the code. I think this "advice" is a bit ill-advised as comments should be part of the document containing the code when using LP. I would probably just drop that last sentence?